Snowflake.Data/Configuration/EasyLoggingConfigFinder.cs

```diff
@@ -43,24 +46,34 @@ private string GetFilePathEnvironmentVariable()
             return GetFilePathFromInputParameter(filePath);
         }
         
-        private string GetFilePathFromTempDirectory() => SearchForConfigInDirectory(Path.GetTempPath());
+        private string GetFilePathFromTempDirectory() => SearchForConfigInDirectory(Path.GetTempPath, "temp");
 
-        private string GetFilePathFromHomeDirectory() => SearchForConfigInDirectory(GetHomeDirectory());
+        private string GetFilePathFromHomeDirectory() => SearchForConfigInDirectory(GetHomeDirectory, "home");
         
         private string GetFilePathFromInputParameter(string filePath) => string.IsNullOrEmpty(filePath) ? null : filePath;
 
         private string GetHomeDirectory() =>_environmentOperations.GetFolderPath(Environment.SpecialFolder.UserProfile);
 
-        private string GetFilePathFromDriverLocation() => SearchForConfigInDirectory(".");
+        private string GetFilePathFromDriverLocation() => SearchForConfigInDirectory(() => ".", "driver");
 
-        private string SearchForConfigInDirectory(string directory)
+        private string SearchForConfigInDirectory(Func<string> directoryProvider, string directoryDescription)
         {
-            if (string.IsNullOrEmpty(directory))
+            try
+            {
+                var directory = directoryProvider.Invoke();
+                if (string.IsNullOrEmpty(directory))
+                {
+                    return null;
+                }
+
+                var filePath = Path.Combine(directory, ClientConfigFileName);
+                return OnlyIfFileExists(filePath);
```

Comment:
What if (3 use cases):
1. file exists but there is no read access?
2. configuration points to folder, which doesn't exist? (eg. cannot create directory in EasyLoggingStarter.Init -> GetLogPath) ?
3. folder exists, but there is no write access for the log file?

Reply:
we don't care about the cases you write about - it will be an error.
If someone makes dummy configuration it is his fault.
I would like just to prevent the driver to fail on on the stage of choosing the configuration file. And only on this stage.